source/simulator/assembly.cc

```diff
@@ -391,11 +394,30 @@ namespace aspect
                                     introspection.n_compositional_fields,
                                     stokes_dofs_per_cell,
                                     parameters.include_melt_transport,
-                                    rebuild_stokes_matrix),
+                                    rebuild_stokes_matrix,
+                                    parameters.use_bfbt),
          internal::Assembly::CopyData::
          StokesPreconditioner<dim> (stokes_dofs_per_cell));
 
     system_preconditioner_matrix.compress(VectorOperation::add);
+    if (parameters.use_bfbt)
+      {
+        inverse_lumped_mass_matrix.compress(VectorOperation::add);
+        IndexSet local_indices=inverse_lumped_mass_matrix.block(0).locally_owned_elements();
```

Comment:
I asked you in a previous review to add spaces around assignment operators ` = `. Here and below they are still missing. You can go through your own pull request and search to these cases here: https://github.com/geodynamics/aspect/pull/5753/files